test/OracleManagerApp.test.js

```diff
@@ -0,0 +1,209 @@
+    dataFeed2 = await DataFeedOracleBase.new()
+    medianDataFeed = await DataFeedOracle.new()
+
+    // initialize oracleManagerApp
+    oracleManagerApp = OracleManagerApp.at(receipt.logs.filter(l => l.event == 'NewAppProxy')[0].args.proxy)
+
+    await dataFeed1.initialize(root)
+    await dataFeed2.initialize(root)
+    await medianDataFeed.initialize([dataFeed1.address, dataFeed2.address ], oracleManagerApp.address)
+  })
+
+  describe('initialize()', () => {
+    it('sets the correct medianDataFeed', async () => {
+      await oracleManagerApp.initialize([root], [dataFeed1.address, dataFeed2.address], medianDataFeed.address)
+      expect(await oracleManagerApp.medianDataFeed()).to.equal(medianDataFeed.address)
+    })
+
+    it('sets the correct approvedDataFeedsLength', async () => {
+      await oracleManagerApp.initialize([root], [dataFeed1.address, dataFeed2.address], medianDataFeed.address)
+      expect((await oracleManagerApp.approvedDataFeedsLength()).toNumber()).to.equal(2)
+    })
+
+    it('sets all dataFeeds to true in approvedDataFeeds mapping', async () => {
+      await oracleManagerApp.initialize([root], [dataFeed1.address, dataFeed2.address], medianDataFeed.address)
+      expect(await oracleManagerApp.approvedDataFeeds(dataFeed1.address)).to.equal(true)
+      expect(await oracleManagerApp.approvedDataFeeds(dataFeed2.address)).to.equal(true)
+    })
+
+    it('sets all permissionsedAccounts to true in permissionedAccounts mapping', async () => {
+      await oracleManagerApp.initialize([root], [dataFeed1.address, dataFeed2.address], medianDataFeed.address)
+      expect(await oracleManagerApp.permissionedAccounts(root)).to.equal(true)
+    })
+
+    it('reverts if duplicate dataFeeds are submitted', async () => {
+      return assertRevert(async () => {
+        await oracleManagerApp.initialize([root], [dataFeed1.address, dataFeed2.address, dataFeed1.address], medianDataFeed.address)
+      })
+    })
+  })
+
+  describe('recordMedianData()', () => {
+    let unapprovedDataFeed
+
+    beforeEach(async () => {
+      unapprovedDataFeed = await DataFeedOracleBase.new()
+      unapprovedDataFeed.initialize(root)
+
+      await oracleManagerApp.initialize([root], [dataFeed1.address, dataFeed2.address], medianDataFeed.address)
+      await dataFeed1.setResult(uintToBytes(5), unixTime())
+      await dataFeed2.setResult(uintToBytes(7), unixTime())
+      await unapprovedDataFeed.setResult(uintToBytes(9), unixTime())
+    })
+
+    it('sets the result to the median of the dataFeeds', async () => {
+      await oracleManagerApp.recordDataMedian([dataFeed1.address, dataFeed2.address])
+      expect(bytes32ToNumString((await medianDataFeed.resultByIndexFor(1))[0])).to.equal(6)
+    })
+
+    it('reverts if a dataFeed is not approved', async () => {
+      return assertRevert(async () => {
+        await oracleManagerApp.recordDataMedian([dataFeed1.address, dataFeed2.address, unapprovedDataFeed.address])
```

Comment:
I think this is hitting `require(dataFeeds.length == approvedDataFeedsLength);` instead of `require(approvedDataFeeds[address(dataFeeds[i])], 'dataFeed is not approved');`